pkg/services/ngalert/api/api_convert_prometheus.go

```diff
@@ -28,9 +59,128 @@ func (srv *ConvertPrometheusSrv) RouteConvertPrometheusGetNamespace(c *contextmo
 }
 
 func (srv *ConvertPrometheusSrv) RouteConvertPrometheusGetRuleGroup(c *contextmodel.ReqContext, namespaceTitle string, group string) response.Response {
-	return response.Error(501, "Not implemented", nil)
+	// Just to make the mimirtool rules load work. It first checks if the group exists, and if the endpoint returns 501 it fails.
+	return response.YAML(http.StatusOK, apimodels.PrometheusRuleGroup{})
 }
 
-func (srv *ConvertPrometheusSrv) RouteConvertPrometheusPostRuleGroup(c *contextmodel.ReqContext, namespaceTitle string, prometheusGroup apimodels.PrometheusRuleGroup) response.Response {
-	return response.Error(501, "Not implemented", nil)
+func (srv *ConvertPrometheusSrv) RouteConvertPrometheusPostRuleGroup(c *contextmodel.ReqContext, namespaceTitle string, promGroup apimodels.PrometheusRuleGroup) response.Response {
+	logger := srv.logger.FromContext(c.Req.Context())
+
+	logger.Info("Converting Prometheus rule group", "folder_title", namespaceTitle, "group", promGroup.Name, "rules", len(promGroup.Rules))
+
+	ns, errResp := srv.getOrCreateNamespace(c, namespaceTitle, logger)
+	if errResp != nil {
+		return errResp
+	}
+
+	logger.Debug("Found folder", "folder_title", ns.Title, "folder_uid", ns.UID)
+
+	datasourceUID := strings.TrimSpace(c.Req.Header.Get(datasourceUIDHeader))
+	if datasourceUID == "" {
+		return response.Error(http.StatusBadRequest, fmt.Sprintf("Missing datasource UID header: %s", datasourceUIDHeader), nil)
+	}
+	ds, err := srv.datasourceCache.GetDatasourceByUID(c.Req.Context(), datasourceUID, c.SignedInUser, c.SkipDSCache)
+	if err != nil {
+		logger.Error("Failed to get datasource", "datasource_uid", datasourceUID, "error", err)
+		return errorToResponse(err)
+	}
+
+	group, err := srv.convertToGrafanaRuleGroup(c, ds, ns.UID, promGroup, logger)
+	if err != nil {
+		return errorToResponse(err)
+	}
+
+	err = srv.alertRuleService.ReplaceRuleGroup(c.Req.Context(), c.SignedInUser, *group, models.ProvenanceConvertedPrometheus)
+	if err != nil {
+		logger.Error("Failed to replace rule group", "folder_title", namespaceTitle, "group", promGroup.Name, "error", err)
+		return errorToResponse(err)
+	}
+
+	return response.JSON(http.StatusAccepted, map[string]string{"status": "success"})
+}
+
+func (srv *ConvertPrometheusSrv) getOrCreateNamespace(c *contextmodel.ReqContext, title string, logger log.Logger) (*folder.Folder, response.Response) {
+	logger.Debug("Getting or creating a new folder", "folder_title", title)
+	ns, err := srv.ruleStore.GetOrCreateNamespaceInRootByTitle(
+		c.Req.Context(),
+		title,
+		c.SignedInUser.GetOrgID(),
+		c.SignedInUser,
```

Comment:
do we need two arguments? perhaps infer OrgID from the user?

Reply:
Kept it consistent with the other methods for the namespaces, but perhaps they all should be changed